sources/main.cpp

```diff
@@ -491,17 +527,35 @@ int main(int argc, const char** argv) try {
         arguments.emplace_back("-resource-dir=" + resource_dir.string());
     }
 
-    //
     // Specify the hyde preprocessor macro
-    // 
     arguments.emplace_back("-DADOBE_TOOL_HYDE=1");
 
+    //
+    // Spin up the tool and run it.
+    //
+
+    ClangTool Tool(OptionsParser.getCompilations(), sourcePaths);
```

Comment:
This is a tweak to how the clang-specific command line arguments are sent into the driver.

We have a major problem here, where the driver will try to helpfully auto-detect the type of file being compiled by the driver, and this setting cannot be overridden. For example, in Photoshop we have a number of header files that end in `.h`, even though they contain C++ code within them. The driver will auto detect them as a C header file, and thus throw errors when c++ flags are passed (e.g., `-std=c++14`). Even trying to force the compilation type to C++ via `-x c++` doesn't work, as it's later overridden in the driver command line arguments by the auto-detect function. (You can see this happening if you run hyde in very verbose mode.)